networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/AttestationTopicSubscriberTest.java

```diff
@@ -81,13 +83,81 @@ public void shouldPreserveLaterSubscriptionPeriodWhenEarlierSlotAdded() {
     final UnsignedLong firstSlot = UnsignedLong.valueOf(10);
     final UnsignedLong secondSlot = UnsignedLong.valueOf(15);
 
-    subscriptions.subscribeToCommitteeForAggregation(subnetId, secondSlot);
-    subscriptions.subscribeToCommitteeForAggregation(subnetId, firstSlot);
+    subscriber.subscribeToCommitteeForAggregation(subnetId, secondSlot);
+    subscriber.subscribeToCommitteeForAggregation(subnetId, firstSlot);
 
-    subscriptions.onSlot(firstSlot.plus(ONE));
+    subscriber.onSlot(firstSlot.plus(ONE));
     verify(eth2Network, never()).unsubscribeFromAttestationSubnetId(subnetId);
 
-    subscriptions.onSlot(secondSlot.plus(ONE));
+    subscriber.onSlot(secondSlot.plus(ONE));
+    verify(eth2Network).unsubscribeFromAttestationSubnetId(subnetId);
+  }
+
+  @Test
+  public void shouldSubscribeToNewSubnetsAndUpdateENR_forPersistentSubscriptions() {
+    Set<SubnetSubscription> subnetSubscriptions =
+        Set.of(
+            new SubnetSubscription(1, UnsignedLong.valueOf(20)),
+            new SubnetSubscription(2, UnsignedLong.valueOf(15)));
+
+    subscriber.subscribeToPersistentSubnets(subnetSubscriptions);
+
+    verify(eth2Network).setLongTermAttestationSubnetSubscriptions(Set.of(1, 2));
+
+    verify(eth2Network).subscribeToAttestationSubnetId(1);
+    verify(eth2Network).subscribeToAttestationSubnetId(2);
+  }
+
+  @Test
+  public void shouldUpdateENRWhenNewSubnetIsSubscribedDueToPersistentSubscriptions() {
+    UnsignedLong someSlot = valueOf(15);
+    Set<SubnetSubscription> subnetSubscription = Set.of(new SubnetSubscription(2, someSlot));
+
+    subscriber.subscribeToCommitteeForAggregation(1, someSlot);
+    subscriber.subscribeToCommitteeForAggregation(2, someSlot);
+
+    verify(eth2Network).subscribeToAttestationSubnetId(1);
+    verify(eth2Network).subscribeToAttestationSubnetId(2);
+
+    subscriber.subscribeToPersistentSubnets(subnetSubscription);
+
+    verify(eth2Network).setLongTermAttestationSubnetSubscriptions(Set.of(2));
+
+    verify(eth2Network).subscribeToAttestationSubnetId(2);
+  }
+
+  @Test
+  public void shouldExtendSubscriptionPeriod_forPersistentSubscriptions() {
+    final int subnetId = 3;
+    final UnsignedLong firstSlot = UnsignedLong.valueOf(10);
+    final UnsignedLong secondSlot = UnsignedLong.valueOf(15);
+    Set<SubnetSubscription> subnetSubscriptions =
+        Set.of(new SubnetSubscription(subnetId, secondSlot));
+
+    subscriber.subscribeToCommitteeForAggregation(subnetId, firstSlot);
+    subscriber.subscribeToPersistentSubnets(subnetSubscriptions);
+
+    subscriber.onSlot(firstSlot.plus(ONE));
+    verify(eth2Network, never()).unsubscribeFromAttestationSubnetId(subnetId);
+
+    subscriber.onSlot(secondSlot.plus(ONE));
+    verify(eth2Network).unsubscribeFromAttestationSubnetId(subnetId);
+  }
+
+  @Test
+  public void shouldPreserveLaterSubscription_forPersistentSubscriptions() {
```

Comment:
May be worth another couple of tests where we subscribe to the same persistent subnet with 2 different unsubscribe slots.